rust-toolchain.toml

```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2024-06-13"
+channel = "stable"
```

Comment:
We need to be careful since `arm` and `riscv*` still need nightly.  Not sure how to set that up to work right, though.

Reply:
I see this, and I'm not sure the best way either. We might need to keep the rust-toolchain pinned and add `+stable` to the non-arm and riscv CI builds. I don't love that solution though... Maybe we should rewrite this to not use the `std` feature detection.

Reply:
It's pretty obscure and bespoke to do it without `std`s feature detection, though.  It didn't feel very stable.

The ones that are stable already do correspond to the tier 1 targets.  Not sure what to do about it either.  I wish `rust-toolchain.toml` could be set per target.

Reply:
I'm going to leave nightly for now and switch CI to stable.